content/articles/gulp4-primer.md

```diff
@@ -0,0 +1,131 @@
+You can have a look at the Goal section of the README to know what this boilerplate does but for those not wanting to open the link: typescript, scss, automatic, DI, template preloading, testing with karma and protractor, live reload.
+Note: I'm not using this boilerplate personally anymore since I don't really do vanilla angular anymore.
+
+Here's the [starting point](https://github.com/Keats/ng-boilerplate/tree/87e75551651e94dfc1aa6135e1ea7cb5bd61cf0f).
+
+## Updating dependencies
+Nothing fancy here, just updating our npm and bower dependencies, including pointing to the gulp4 branch on github.
+To use npm install from a github repo, you can use:
+```
+$ npm install --save-dev gulpjs/gulp.git#4.0
+```
+Tests are going to fail at that point because of changes in the api of gulp and probably some in other libraries we just updated as well.
+You can see the commit [here](https://github.com/Keats/ng-boilerplate/commit/49a339cc5cbde8f5374b8f4c915a548dc2916cc4).
+
+## Migrating the gulpfile
+### Calling gulp
+The first thing we need to do is ensure that we are using the right version of gulp.  You might have another version installed globally or nothing installed globally like me.
+The easiest way (at least to me) to deal with that issue is the [npm scripts](https://docs.npmjs.com/misc/scripts) part of the package.json.
+
+```js
+  "scripts": {
+    "gulp": "./node_modules/gulp/bin/gulp.js"
+  },
+```
+The downside is that you will get npm errors instead of the regular one but that's only an issue when making your gulpfile.
+Let's not forget to update the travis.yml to use that npm script too.
+You can see the commit [here](https://github.com/Keats/ng-boilerplate/commit/ef30315f43580e8b3c1ba85d4536c7fe0d69365d).
+
+### Fixing the errors we're getting
+If you run the gulpfile locally or look at the [travis build](https://travis-ci.org/Keats/ng-boilerplate/builds/54252789), you will notice we get the following error:
+```bash
+assert.js:87
+  throw new assert.AssertionError({
+        ^
+AssertionError: Task function must be specified
+    at Gulp.set (/home/vincent/Code/ng-boilerplate/node_modules/gulp/node_modules/undertaker/lib/set.js:14:3)
+    at Gulp.task (/home/vincent/Code/ng-boilerplate/node_modules/gulp/node_modules/undertaker/lib/task.js:14:8)
+....
+```
+This comes from the fact that gulp.task changed and the 3 parameters signature was removed.
+In short, tasks that look like the one below will need to be changed to remove the [] parameter:
+```js
+gulp.task('default', ['build'], function () {
+  return runSequence(['watch', 'karma-watch']);
+});
+```
+To give a quick background to people not familiar to gulp, the list parameter is a list of tasks that need to be ran before running that task, in this case it will call the build task before running watch and karma-watch in order (runSequence is another thing that can be changed but we will get back to that later).
+To solve that very common issue of task dependencies, 2 methods were added:
```

Comment:
In English it's common to write out numerals up to 5 or so. So use "two methods" were added